src/core/border_router/routing_manager.cpp

```diff
@@ -817,6 +817,45 @@ bool RoutingManager::NetworkDataContainsUlaRoute(void) const
     return contains;
 }
 
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_REACHABLITY_CHECK_ICMP6_ERROR_ENABLE
+
+void RoutingManager::CheckReachbilityToSendIcmpError(const Message &aMessage, const Ip6::Header &aIp6Header)
+{
+    Ip6::MessageInfo messageInfo;
+
+    VerifyOrExit(IsRunning() && IsInitalPolicyEvaluationDone());
+
+    VerifyOrExit(!aIp6Header.GetDestination().IsMulticast());
+
+    // Validate that source matches the ULA OMR prefix (which is not
+    // infrastructure-derived).
+
+    VerifyOrExit(!mOmrPrefixManager.GetFavoredPrefix().IsEmpty());
+    VerifyOrExit(!mOmrPrefixManager.GetFavoredPrefix().IsInfrastructureDerived());
+    VerifyOrExit(aIp6Header.GetSource().MatchesPrefix(mOmrPrefixManager.GetFavoredPrefix().GetPrefix()));
+
+    VerifyOrExit(mRoutePublisher.IsPublishingDefaultRoute());
+
+    VerifyOrExit(!mRxRaTracker.IsAddressOnLink(aIp6Header.GetDestination()));
+    VerifyOrExit(!mRxRaTracker.IsAddressReachableThroughExplicitRoute(aIp6Header.GetDestination()));
+    VerifyOrExit(!Get<NetworkData::Leader>().IsNat64(aIp6Header.GetDestination()));
+
+    LogInfo("Send ICMP unreachable for fwd msg with local ULA src and outside-AIL dst");
```

Comment:
"outside-AIL dst" is somewhat ambiguous here - e.g. the destination could be another Thread Device in another network connected to the AIL. This could be viewed as "outside AIL" but still reachable.  But I don't have a better term at the moment. Spec text is also incorrect on this detail.

Reply:
Yes. For the log, I wanted some short text to convey the message. Please let me know if you can think of better name.

Reply:
based on the new proposed spec text in SPEC-1312, the following would be slightly better:

```suggestion
    LogInfo("Send ICMP unreachable for fwd msg with local ULA src and non-local dst");
```

Reply:
Thanks. Changed in new push.